core/src/price_estimation/average_price_source.rs

```diff
@@ -6,6 +6,7 @@ use std::collections::HashMap;
 use std::hash::Hash;
 use std::num::NonZeroU128;
 
+/// Owns inner price sources and applies `average_price_sources` to them in `get_prices`.
```

Comment:
ubernit: I'm not a fan of this comment as I don't see how it gives additional non-obvious information and is likely going to get outdated if anything about the methods below changes.